flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -590,6 +604,15 @@ public Execution resetForNewExecution(final long timestamp, final long originati
 
 				this.currentExecution = newExecution;
 
+				synchronized (this.inputSplits){
```

Comment:
I wonder if the synchronization on `inputSplits` is even needed? In the failover case, `synchronized (priorExecutions)` makes failing exclusive and failing checks that the previous attempt is no longer running before manipulating the input splits. Getting the next split should then only be called from one attempt at a time, once its running and that should be from a single thread. Did I miss something?

Reply:
The member `inputSplits` will be accessed in another method `getNextInputSplit()` which may be invoked in another thread. I would like to suggest to keep this safe guard here.